Il2CppInterop.Runtime/Injection/AssemblyList/GameManagersAssemblyListFile.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Il2CppInterop.Runtime.Injection
+{
+    internal class GameManagersAssemblyListFile : IAssemblyListFile
+    {
+        private List<string> _assemblies = new List<string>();
+        private string originalFile;
+        private string newFile;
+
+        public bool IsTargetFile(string originalFilePath)
+        {
+            return originalFilePath.Contains("globalgamemanagers");
+        }
+
+        public void Setup(string originalFilePath)
+        {
+            if (originalFile != null) return;
+            originalFile = originalFilePath;
+        }
+
+        public void AddAssembly(string name)
+        {
+            _assemblies.Add(name);
+        }
+
+        public string GetOrCreateNewFile()
+        {
+            if (newFile != null) return newFile;
+
+            newFile = Path.GetTempFileName();
+            CreateModifiedFile();
+            return newFile;
+        }
+
+        private void CreateModifiedFile()
```

Comment:
I am strongly against a solution that requires modifying game files.

Reply:
Even if we don't actually touch original files? This PR uses hooks at runtime to swap files when the game requests them. Nothing is modified on disk.